src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpDiagnosticIdToEditorConfigOptionMappingService.cs

```diff
@@ -0,0 +1,93 @@
+                case IDEDiagnosticIds.UseExpressionBodyForPropertiesDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferExpressionBodiedProperties;
+                    break;
+                case IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferExpressionBodiedIndexers;
+                    break;
+                case IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferExpressionBodiedAccessors;
+                    break;
+                case IDEDiagnosticIds.UseDefaultLiteralDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferSimpleDefaultExpression;
+                    break;
+                case IDEDiagnosticIds.UseLocalFunctionDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferLocalOverAnonymousFunction;
+                    break;
+                case IDEDiagnosticIds.InvokeDelegateWithConditionalAccessId:
+                    option = CSharpCodeStyleOptions.PreferConditionalDelegateCall;
+                    break;
+                case IDEDiagnosticIds.UseExpressionBodyForLambdaExpressionsDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferExpressionBodiedLambdas;
+                    break;
+                case IDEDiagnosticIds.UseIndexOperatorDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferIndexOperator;
+                    break;
+                case IDEDiagnosticIds.UseRangeOperatorDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferRangeOperator;
+                    break;
+                case IDEDiagnosticIds.ExpressionValueIsUnusedDiagnosticId:
+                    option = CSharpCodeStyleOptions.UnusedValueExpressionStatement;
+                    break;
+                case IDEDiagnosticIds.ValueAssignedIsUnusedDiagnosticId:
+                    option = CSharpCodeStyleOptions.UnusedValueAssignment;
+                    break;
+                case IDEDiagnosticIds.UseExpressionBodyForLocalFunctionsDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferExpressionBodiedLocalFunctions;
+                    break;
+                case IDEDiagnosticIds.MakeLocalFunctionStaticDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferStaticLocalFunction;
+                    break;
+                case IDEDiagnosticIds.UseSimpleUsingStatementDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferSimpleUsingStatement;
+                    break;
+                case IDEDiagnosticIds.MoveMisplacedUsingDirectivesDiagnosticId:
+                    option = CSharpCodeStyleOptions.PreferredUsingDirectivePlacement;
+                    break;
+
+                default:
+                    return (optionOpt: null, handled: false);
+            }
+
+            return (optionOpt: option, handled: true);
```

Comment:
this is *super* ick... I'm very very opposed to the idea that we add new diagnostic features, but then have to update a dozen or so independent parts of the IDE codebase in order for them to "light up" all these other features properly.  I am strongly pushing that we stop continually doing this and we go back and reassess how DiagnosticAnalyzers work in order to have them properly enhance the IDE experience.

For example, perhaps there can be a new interface that a DiagnosticAnalyzer optionally implements (or no-op virtual methods they override).  These can supply this additional information that these auxilliary features need.

Reply:
This is an interim solution, I completely agree we need a better solution, in fact something that works even for third party analyzers who want to be configurable through .editorconfig code style options. However, that is a much bigger work item. I have still ensured that for the current approach we have unit tests that ensure that every new IDE diagnosticID is handled explicitly here.  If someone adds a new IDE diagnostic ID, added unit test will fail with a message guiding how to handle it correctly.

> perhaps there can be a new interface that a DiagnosticAnalyzer optionally implements 

This approach does not help here. We need a generic mapping from a reported diagnostic/diagnosticID to corresponding code style option used to configure it, so that any configuration code action can tweak this option in analyzer config document.